### PR TITLE
#20

### PR DESCRIPTION
**Pull request by sdehgh**
update category controler
update API document collection for update

### CODE REVIEW

File: app/Http/Controllers/API/CategoryController.php [hunk 1]
```diff
@@ -39,17 +39,20 @@ public function store(CategoryRequest $request): \Illuminate\Http\JsonResponse
...
      * @param CategoryRequest $request
      * @param \App\Models\Category $category
-     * @return \Illuminate\Http\Response
+     * @return \Illuminate\Http\JsonResponse
      */
     public function update(CategoryRequest $request, Category $category)
     {
-        //
+        Category::update($request->validated());
+        $this->getCategoryInTree()->update($category,$request);
```
Comment (mj999666) [inline app/Http/Controllers/API/CategoryController.php]: What is this?
For what purpose did you do this?
If you do not have a specific goal, delete this line

File: routes/api.php [hunk 2]
```diff
@@ -27,3 +27,7 @@
     ->except("index")
     ->middleware('auth:sanctum');
 Route::get("/categories" , [\App\Http\Controllers\API\CategoryController::class , "index"])->name("categories.index");
+
+
+
+Route::put("categories/{id}",[\App\Http\Controllers\API\CategoryController::class , "update"])->name("categories.update");
```
Comment (mj999666) [inline routes/api.php]: You don't need this
If you look at the top line, we have this
```
Route::apiResource("/categories" , \App\Http\Controllers\API\CategoryController::class)
    ->except("index")
    ->middleware('auth:sanctum');
```
There are also update items in `apiResource`
Use `php artisan route:list` to see the list of routes and research about `apiResource`

**Review by mj999666 — COMMENTED** [review]
Please check the items mentioned in the comments

File: app/Http/Controllers/API/CategoryController.php (hunk 1 above)
Comment (sdehgh) [reply app/Http/Controllers/API/CategoryController.php]: I meant that the tree should be updated based on category and request

File: routes/api.php (hunk 2 above)
Comment (sdehgh) [reply routes/api.php]: ok,thanks.

File: app/Http/Controllers/API/CategoryController.php [hunk 3]
```diff
@@ -39,17 +39,20 @@ public function store(CategoryRequest $request): \Illuminate\Http\JsonResponse
             "categories" => new CategoryCollection($this->getCategoryInTree()) ,
         ] , 201);
     }
-
     /**
      * Update the specified resource in storage.
      *
      * @param CategoryRequest $request
      * @param \App\Models\Category $category
-     * @return \Illuminate\Http\Response
```
Comment (mj999666) [inline app/Http/Controllers/API/CategoryController.php:48]: remove this line
your return type is `JsonResponse`

**Comment by sdehgh** [comment]
I closed this pull request so that I can request the changes again to 17-category-document.

**Pull request closed without merging**